src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -70,6 +72,32 @@ static PowerShellKernel()
             // Add accelerators that exist in other namespaces.
             addAccelerator.Invoke(null, new object[] { "Layout", typeof(Layout.Layout) });
             addAccelerator.Invoke(null, new object[] { "Chart", typeof(Chart) });
+
+            //TODO: Is this declared anywhere else?
+            string[] mimeTypes = new string[] { HtmlFormatter.MimeType, JsonFormatter.MimeType, PlainTextFormatter.MimeType };
+
+            foreach (string mt in mimeTypes)
+            {
+                Formatter<PSObject>.Register((psObj, writer) =>
+                {
+                    object obj;
+                    if (psObj.BaseObject is PSCustomObject)
+                    {
+                        Dictionary<string, object> table = new Dictionary<string, object>();
+                        var props = psObj.Properties;
+                        foreach (var p in props)
+                        {
+                            table.Add(p.Name, p.Value);
+                        }
+                        obj = table;
+                    }
+                    else
+                    {
+                        obj = psObj.BaseObject;
+                    }
+                    obj.FormatTo(writer, mt);
+                }, mimeType: mt);
```

Comment:
Why moving this here? I think it fits better in the `Out-Display` command.
This formatting doesn't do anything special that cannot be done in general before you call `display` in `Out-Display`.

Reply:
Let's fix the `PSCustomObject` only in this PR and defer the fix for all ETS extended properties to a separate PR.

Reply:
Why do it in Out_Display when .NET Interactive already has this Formatter concept? It seems like we should leverage that instead.

Reply:
I don't see any benefit to use the formatter for the current code. It's fine to do this if it's a fix for PSObject in general, but it's not.
Actually, now I double the formatter is the right tool to solve the PSObject ETS issue in general.
Besides, do you know where those formatter will be invoked? Maybe on a different thread that doesn't have a default Runspace, then `p.Value` will fail if the property is a `ScriptProperty`.

Reply:
Reread the code - 100% agree. @justinytchen and I synced on this and we're all on the same page.

Reply:
Great! Also, we we do need a formatter for anything, then it's better to do it using the extension method to `Kernel`, just like how we did for invoking the profile.

Reply:
@daxian-dbw I didn't see your longer comment before commenting but exactly. Justin's gonna change this PR back but we'll defer the generic solution for now. He's gonna focus on `Out-GridView` :)

Reply:
That's right, focus on `Out-GridView`. The PSObject ETS member issue is a _hard and not urgent_ one to fix.